misk/src/main/kotlin/misk/inject/KAbstractModule.kt

```diff
@@ -0,0 +1,46 @@
+package misk.inject
+
+import com.google.inject.AbstractModule
+import com.google.inject.binder.AnnotatedBindingBuilder
+import com.google.inject.binder.LinkedBindingBuilder
+import com.google.inject.binder.ScopedBindingBuilder
+import com.google.inject.multibindings.Multibinder
+
+/**
+ * A class that provides helper methods for working with Kotlin and Guice, allowing implementing
+ * classes to operate in the Kotlin type system rather than converting to Java.
+ *
+ * The more Kotlin friendly API allows calls like:
+ *
+ * ```
+ * bind(Foo::class.java).to(RealFoo::class.java)
+ * ```
+ *
+ * To be rewritten as:
+ * ```
+ * bind<Foo>().to<RealFoo>()
+ * ```
+ */
+abstract class KAbstractModule : AbstractModule() {
```

Comment:
I don’t think this needs to be a type, just some extension functions? Ie. it doesn’t need state?

Reply:
It needs to extend `AbstractModule` because `binder` is protected.